rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+This is not meant to be indicative of what should happen or not, but indicative of what could happen.
+Each package's individual `license` field setup is left to its maintainers, and nixpkgs governance should conflict arise.
+This RFC does not mean to indicate that it is right or wrong, and is not the right place to discuss changes to this field.
+Should one have disagreements on any specific package in this list, please bring that up to that package's maintainers.
+
+It is also suggested in this RFC that people, upon marking licenses as `runnableOnHydra`, check all the derivations that use this license.
+They could then have to mark them as either `hydraPlatforms = []`, `preferLocalBuild = true` and/or `allowSubstitutes = false`.
+This might be useful for packages like TPTP:
+they may not yet be marked as such due to these flags having no impact on unfree packages;
+but would take gigabytes on Hydra for basically no local build time improvement
+
+With this in mind, Hydra could start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
+- Vagrant
+- NixOS tests that involve such software (eg. MongoDB or Nomad)
+
+And Hydra will keep not building, among others:
+- CompCert
+- DataBricks
+- Elasticsearch
+- GeoGebra
+- Widevine CDM
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Technical drawbacks
+
+The main risk is that NixOS could end up including unfree software in an installation image if:
+1. we forgot to add it to the list of no-allowed-unfree jobs, and
+2. a maintainer did actually add unfree software to that build.
+
+This seems exceedingly unlikely, making this change basically risk-free.
+
+The only remaining drawback is that Hydra would have to evaluate Nixpkgs twice, thus adding to eval times.
+However, the second eval (with no-unfree) should be reasonably small and not actually evaluate all packages, as it is only used for installation media.
+
+## Political drawbacks
+
+Whether distributing unfree software is a political drawback is left to each reader's opinion.
+
+Besides that, there are three main political risks.
+First is, this RFC could end up completely unused.
+Maybe, with proper license investigation, we will notice that none of the packages listed above can actually be redistributed by Hydra.
+
+The second risk is one of manpower.
+We may need the Foundation's input on whether a specific license is ok to redistribute or not.
+This could require some manpower from the Foundation's side.
+
+Finally, the third risk is one of propagation.
+With both hydra and some nixos maintainers running with `allowUnfree`, there is a risk that free packages start unnecessarily depending on unfree packages.
+This would then break the setup of the people not actually running with `allowUnfree`.
+
+This being said, all these risks are probably less impactful than the current statu quo.
```

Comment:
Can you explain this? I don’t think the status quo is risky at all, even if you prefer this change. It can be inconvenient – I guess we could quantify that as a risk of making NixOS less popular?

Reply:
It's explained by the rest of the paragraph in which this line is: basically, it's the darwin packages example where people can end up having nonfree package in their nix store even though they never actually set ALLOW_UNFREE